### ISSUE TITLE
[bug] API - "Get all datasources" doesn't return jsonData even if present

### ISSUE THREAD

**Issue by rmoff** [grafana/grafana#4340]
Grafana v 2.6.0

If I call [get all data sources](http://docs.grafana.org/reference/http_api/#get-all-datasources) each datasource is returned, but with `"jsonData": null`. If I call [get datasource by id](http://docs.grafana.org/reference/http_api/#get-a-single-data-sources-by-id) then it (correctly) includes jsonData. 

Sample requests & responses: 

```bash
[oracle@demo grafana]$ curl -X GET -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "Cache-Control: no-cache" "http://192.168.56.101:3000/api/datasources"
```

```json
[{"id":15,"orgId":1,"name":"Elasticsearch - ASH","type":"elasticsearch","access":"proxy","url":"http://localhost:9200","password":"","user":"","database":"[logstash-ash]-YYYY-MM-DD","basicAuth":false,"basicAuthUser":"","basicAuthPassword":"","withCredentials":false,"isDefault":false,"jsonData":null}]
```

```bash
[oracle@demo grafana]$ curl -X GET -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "Cache-Control: no-cache" "http://192.168.56.101:3000/api/datasources/15"
```

```json
{"id":15,"orgId":1,"name":"Elasticsearch - ASH","type":"elasticsearch","access":"proxy","url":"http://localhost:9200","password":"","user":"","database":"[logstash-ash]-YYYY-MM-DD","basicAuth":false,"basicAuthUser":"","basicAuthPassword":"","withCredentials":false,"isDefault":false,"jsonData":{"interval":"Daily","timeField":"@timestamp"}}
```

**Comment by torkelo** [comment]
this is by design, you only get that when requesting a specific data source

**Comment by rmoff** [comment]
Why is that? And, should it be noted in the docs? 
thanks!

**Comment by torkelo** [comment]
Because list apis that return many items should not returning  full data (especially not for fields that can be very big) , think the http api is correct , or is it showing with json data?

**Comment by torkelo** [comment]
I mean http api docs is correct

**Comment by rmoff** [comment]
The doc doesn't mention that the data returned won't include all fields
(I.e. Is partial), and the get all api does return a value for jsondata
(null) rather than omitting the field that would imply that it is there to
be fetched on request.

**Comment by torkelo** [comment]
oh, that is right, the fact that it was null was a bug and is fixed in master / upcoming Grafana 3.0
